src/main/java/com/google/sps/data/Contest.java

```diff
@@ -83,6 +93,30 @@ public static Contest fromJSONObject(DatastoreService datastore, JSONObject cont
         .build();
   }
 
+  // Convert this Contest object to a String in JSON format. Serializes its list of Candiate Entity
+  // key names into JSON objects that correspond to each Candidate's attributes.
+  public String toJsonString(DatastoreService datastore) {
+    Gson gson = new Gson();
+    JsonArray candidateJsonArray = new JsonArray();
+
+    // Since a Contest object's "candidates" collection consists only of their Datastore key names,
```

Comment:
Did you mean Datastore ids? you call the variable "contestCandidateIds" below :P

Reply:
Oop that should be "key name" all around, actually. Will update!

Reply:
Wait jk it's ID